User Sign Up/Profile Flow

emjayholla opened the pull request:
- Create account is functioning. 
- User profile fields are now editable and will save in the database. 
- SQL files have been changed, *be sure to drop the user_profile table and re-run it*

emjayholla committed:
updating after merge.

emjayholla committed:
Adding changes pulled from Main.

emjayholla committed:
Dalton's backend code added.

emjayholla committed:
Began conversion over to Hook Form.

emjayholla committed:
small changes

emjayholla committed:
small commit to prepare for merge.

emjayholla committed:
completed merge

emjayholla committed:
updated fields.

emjayholla committed:
Began validation on profile page.

emjayholla committed:
need to pull

emjayholla committed:
Merge branch 'main' into maggie-new-branch

emjayholla committed:
minor changes to sql.

emjayholla committed:
created sign up page and routing to edit profile page.

emjayholla committed:
small changes to hook form.

emjayholla committed:
merged in hook form.

emjayholla committed:
created working sign up page, sign up button routes to user profile that can be edited.

LaplacesAngel committed:
Merge pull request #32 from mrao2/paul-navbar-3

CSS updates

brookebrown617 committed:
added

Dalton-Kujawa committed:
Fixed the css for the web pages

LaplacesAngel committed:
accepeted your chanees in the login page and my changes in navbar

emjayholla committed:
used newly created id and passed it into the URL. Not ready to merge yet.

LaplacesAngel committed:
updated index.js on the backend and got some data to display from my backend table

brookebrown617 committed:
deleted confusing sql file

brookebrown617 committed:
changes

LaplacesAngel committed:
navbar and homepage updates

brookebrown617 committed:
Merge pull request #31 from mrao2/brooke-login3

added sign up button that links to profile

LaplacesAngel committed:
Merge branch 'main' into 5-homepage-and-about-us

brookebrown617 committed:
sign up styling

brookebrown617 committed:
minor styling changes

brookebrown617 committed:
hashing function and more login logic

Dalton-Kujawa committed:
Merge pull request #33 from mrao2/Dalton-Add-Comments

Fixed the css for the web pages

LaplacesAngel committed:
Merge branch 'main' into 5-homepage-and-about-us

brookebrown617 committed:
changes

brookebrown617 committed:
deleted axios

emjayholla committed:
fixed button bugs.

emjayholla committed:
small fixes.

brookebrown617 committed:
did what i needed to do

brookebrown617 committed:
whoops

brookebrown617 committed:
Merge pull request #34 from mrao2/brooke-login4

Brooke login4

LaplacesAngel committed:
deleted node modules and package lock

LaplacesAngel committed:
Merge branch 'main' into 5-homepage-and-about-us

emjayholla committed:
deleted node modules and package-lock/package.

emjayholla committed:
.

emjayholla committed:
.

emjayholla committed:
pulled from main.

LaplacesAngel committed:
added dockerfiles and messed with bcrypt stuff, doesnt work yet

LaplacesAngel committed:
clean up package lock shit

LaplacesAngel committed:
uncommented bcrypt stuff

LaplacesAngel committed:
updated frontend package lock to version 3

Dalton-Kujawa committed:
Fixed the issue with blogs if no comments it would not load

Dalton-Kujawa committed:
Merge pull request #36 from mrao2/5-homepage-and-about-us

have some homepage updates, mainly a packagelock.json fix

LaplacesAngel committed:
updated navbar title

LaplacesAngel committed:
Merge branch 'main' of https://github.com/mrao2/newtowndogs

LaplacesAngel committed:
Merge branch 'main' into 5-homepage-and-about-us

LaplacesAngel committed:
npm i changes

LaplacesAngel committed:
updated name of site to reflect what the business actually does

Dalton-Kujawa committed:
Added CRUD operations for comments now

Dalton-Kujawa committed:
Added the ability to Create comments

Dalton-Kujawa committed:
Merge branch 'main' into Dalton-Add-Comments

Dalton-Kujawa committed:
Merged with main succesfully, everything is still working

Dalton-Kujawa committed:
Fixed the create API endpoint

Dalton-Kujawa committed:
Fixed an issue with my Table sql, and renamed a few variables that had spelling errors

LaplacesAngel committed:
got rid of the run command in docker file

Dalton-Kujawa committed:
Merge pull request #37 from mrao2/Dalton-Add-Comments

LaplacesAngel committed:
Merge branch 'main' of https://github.com/mrao2/newtowndogs

LaplacesAngel committed:
Merge branch 'main' into 5-homepage-and-about-us

LaplacesAngel committed:
comments create statments, added ON DELETE CASCADE

LaplacesAngel committed:
Merge pull request #38 from mrao2/5-homepage-and-about-us

updated sql statements to delete blogs with comments, added a docker file, and some small home page and navbar changes

emjayholla committed:
Merge branch 'main' of https://github.com/mrao2/newtowndogs

emjayholla committed:
Merge branch 'main' into maggie-signup

emjayholla committed:
Create account/save user profile

- Create account is functioning.
-User profile fields are now editable and will save in the database.
- SQL files have been change, *be sure to drop the user_profile table and re-run it*

emjayholla committed:
Cleaned up console logs and commented out code

emjayholla committed:
syntax issue fixed

emjayholla committed:
Update .gitignore

emjayholla committed:
Merge branch 'main' into maggie-signup

LaplacesAngel reviewed (commented):
For some reason can't get the submit button to work after hitting signup:
![image](https://github.com/mrao2/newtowndogs/assets/118093209/4d43cebf-ef80-4221-a6b0-850a2ed589f4)


Also not able to load the profile page:
![image](https://github.com/mrao2/newtowndogs/assets/118093209/bc2c3128-53f9-4b4c-bae4-a90d2711edeb)


Not sure if it's just me or not, maybe someone else could take a look? 
Or Maggie are there potentially changes on your local branch that haven't been pushed up yet if it's working for you?

emjayholla committed:
Update 2_user_profile.sql

Dalton-Kujawa commented:
Hey Maggie I could create a user through sign up but is the profile page support to work at all? its crashing the back end due to some table joining issues.

emjayholla committed:
fixed create function

emjayholla committed:
Merge branch 'main' into maggie-signup

emjayholla committed:
merge fix

emjayholla committed:
did npm i in frontend and backend

emjayholla committed:
CSS

matched 'sign up' button and 'save changes' button to navbar links.

emjayholla committed:
More small CSS changes

made it look pretty

Dalton-Kujawa reviewed (changes requested):
Hey Maggie, things are working well and look good on both pages. I have to request changes only because both sign up and profile need some changes before we can commit them into main. They both need error handling on screen. As it is right now if a user puts in a field that isnt allowed (aka. A phone number thats already used, or a putting in Texas instead of TX, or using too many characters for some names) it will crash the backend. I think just a little bit of verification on the front end input should fix that right up! Also in .vscode/settings.json you have eslint.enable set to false. This would get rid of all those nice red squiggles that tell use when we are making mistakes in our code. ( although they can be annoying)

Dalton-Kujawa commented on backend/data/db.json:
```diff
@@ -0,0 +1,21 @@
+{
```
I have no idea where this file came from lol this is from my original push weeks ago!

emjayholla committed:
added validation to signup.js

emjayholla committed:
Fixed try/catch on /api/sign-up

index.js shows more changes because of the indention fixes, but I only changed the post for sign up and the get for /profile/:id

emjayholla committed:
more validation added to phone and email

emjayholla committed:
.

Dalton-Kujawa reviewed (approved):
![BackendErrorCrash](https://github.com/mrao2/newtowndogs/assets/125609395/8acdf4eb-8d3b-4516-9af8-c7175b1dd2e4)
I am going to approve the code but I did get a backend crash for some of the inputs on profile. If you can make sure you pull from main before you Merge the branch in and make sure you dont have any conflicts! Thanks!

emjayholla committed:
Merge branch 'main' into maggie-signup

emjayholla committed:
pushing after merge.
